lib/app_component.dart

```diff
@@ -18,18 +19,25 @@ import 'todo_list/todo_list_component.dart';
   providers: const [materialProviders],
 )
 class AppComponent implements OnInit{
-  String date;
+  AppComponent() {
+    dateChanged.listen((val) => date = val);
+  }
+
+  @ViewChild('datepicker')
+  ElementRef inputBox;
+
+  String date = '1';
+
+  final StreamController<String> _dateChangedCtrl = new StreamController.broadcast();
+
+  Stream<String> get dateChanged => _dateChangedCtrl.stream;
 
   @override
   ngOnInit() {
-    var function = new js.JsFunction.withThis((input,val,object) {
-      date = val;
-      print(input);
-      print(val);
-      print(object);
+    var fun = new js.JsFunction.withThis((input, val, object) {
+      _dateChangedCtrl.add(val);
     });
-    js.context.callMethod(r'$', ['#datepicker'])
-        .callMethod('datepicker', [new js.JsObject.jsify({'onSelect':function})]);
+    js.context.callMethod(r'$', [inputBox.nativeElement])
```

Comment:
I used the element referenced with `#datepicker` instead using the id. This has nothing to do with the event handling  is just to avoid errors in the future when you have multiple elements with the same id.